### PR TITLE
[Merged by Bors] - syncer: add timeout to layer polling

### PR DESCRIPTION
**Pull request by countvonzero**
## Motivation

related to #3444
during network partition test, occasionally observed that `syncLayer` hangs and never return.

## Changes

- add layer polling timeout

### CODE REVIEW

File: syncer/data_fetch.go [hunk 1]
```diff
@@ -13,6 +14,8 @@ import (
 	"github.com/spacemeshos/go-spacemesh/rand"
 )
 
+const pollTimeOut = time.Second * 30
```
Comment (dshulyak) [inline syncer/data_fetch.go]: could you please use 30 * time.Second, it is common in golang to write number before duration type

**Comment by dshulyak** [comment]
do you know what hangs in the stack? every p2p server is configured with RequestTimeout in the fetcher, default value is 10s. so it could be that this timeout won't help as well

**Comment by countvonzero** [comment]
> do you know what hangs in the stack? every p2p server is configured with RequestTimeout in the fetcher, default value is 10s. so it could be that this timeout won't help as well

there are 2 levels of waiting on top of that
- polling across N peers for each layer
- multiple requests to p2p server for each layer
  - polling layer data (ballot isd and block ids)
  - fetch those ids
 
most likely the bug is in the last one (fetching ids, see #3444). the batching of hash requests were not updated since i joined and i didn't invest time to debug there. 

my motivation for this PR are
- clients of the fetch library enforce a timeout so data syncing can progress
- print a warning when fetching times out for further investigation

**Review by dshulyak — APPROVED** [review]
could be also leaking requests, definitely worth investigating

it should be possible to drop batching altogether, and use libp2p stream per request. streams are not expensive to create

**Comment by countvonzero** [comment]
bors merge

**Pull request closed without merging**